### ISSUE TITLE
Unoptimal solution

### ISSUE THREAD

**Issue by Lightningblade** [jackowski626/universal_tc_research_solver#4]
![image](https://user-images.githubusercontent.com/30607152/109294287-461b7c80-782d-11eb-825c-a50cd8eae7f2.png)

can be reduced to 

![image](https://user-images.githubusercontent.com/30607152/109294301-4ca9f400-782d-11eb-82be-83a24de627e0.png)

Elements excluded: Gelum, Tempus, Volatus
Mods enabled: Vanilla TC, Forbidden Magic, Avaritia, Thaumic Revelations

Console:
https://gist.github.com/Lightningblade/2ce3acf124ab98e271b4dbb10e4cd0ea

**Comment by Lightningblade** [comment]
Another example. Same setup.
![image](https://user-images.githubusercontent.com/30607152/109296914-68af9480-7831-11eb-83c9-5d9b0ae3c5f7.png)
![image](https://user-images.githubusercontent.com/30607152/109296924-6c431b80-7831-11eb-8772-1e5efaf236b7.png)

**Comment by jackowski626** [comment]
Yep I had issues with the "optimising" algorithm a while back too. Gonna see
